pkgs/development/python-modules/ankipandas/default.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, numpy
+, pandas
+, colorlog
+, pytest
+, pytest-cov
+, randomfiletree
+}:
+
+buildPythonPackage rec {
+  pname = "ankipandas";
+  version = "0.3.15";
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-31kkXytwUHeua6TBZMyfuvmQa9KNKcK/ielwDqNj37w=";
+  };
+
+  propagatedBuildInputs = [
+    numpy
+    pandas
+    colorlog
+  ];
+
+  pythonImportsCheck = [ "ankipandas" ];
+
+  nativeCheckInputs = [ pytest pytest-cov randomfiletree ];
+  doCheck = true;
+  preCheck = ''
+  # Resolve OSError: [Errno 24] Too many open files: '/build/tmpr5xzzy53'
+  ulimit -n 2048
+  '';
+  checkPhase = ''
+    runHook preCheck
+    pytest
+    runHook postCheck
+  '';
```

Comment:
If you put `pytestCheckHook` into `nativeCheckInputs` instead of `pytest`, it will auto-run it, so no need for this custom `checkPhase`. `doCheck` will also be able to be removed.